scripts/tests/upload_rcpsnaps.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# This script compresses all `.rpcsnap.json` files in a given directory using zstd,
+# uploads the compressed files to our DigitalOcean Spaces,
+# updates `test_snapshots.txt` with the uploaded filenames,
+# and prompts the user to optionally run the regression tests.
+
+SPACE_NAME="forest-snapshots"
+DEST_DIR="rpc_test"
+
+TEST_SNAPSHOTS="src/tool/subcommands/api_cmd/test_snapshots.txt"
+
+if [ -z "$1" ]; then
+    echo "❌ Please provide the directory path as an argument."
+    echo "Usage: $0 <directory_path>"
+    exit 1
+fi
+
+DIR_PATH="$1"
+
+if [ ! -d "$DIR_PATH" ]; then
+    echo "❌ Directory does not exist: ${DIR_PATH}"
+    exit 1
+fi
+
+for FILE_PATH in "${DIR_PATH}"/*.rpcsnap.json; do
+    if [ ! -f "$FILE_PATH" ]; then
+        echo "❌ No .rpcsnap.json files found in ${DIR_PATH}"
+        break
+    fi
+
+    FILE_NAME=$(basename "$FILE_PATH")
+    COMPRESSED_FILE="${FILE_PATH}.zst"
+    DEST_PATH="${DEST_DIR}/${FILE_NAME}.zst"
+    BUCKET_URL="s3://${SPACE_NAME}/${DEST_PATH}"
+
+    if zstd -f "$FILE_PATH" -o "$COMPRESSED_FILE"; then
```

Comment:
In general, can we simplify things via `set -euo pipefail`? I think error handling would be better this way.